flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/main/java/org/apache/flink/cdc/connectors/postgres/source/PostgresDialect.java

```diff
@@ -217,6 +220,11 @@ public JdbcSourceFetchTaskContext createFetchTaskContext(JdbcSourceConfig taskSo
 
     @Override
     public void notifyCheckpointComplete(long checkpointId, Offset offset) throws Exception {
```

Comment:
Yes, maybe PostgresSourceReader is good option.
I didn't want to have it in IncrementalSourceReaderWithCommit since this is only PG related feature.